bundles/org.eclipse.equinox.common/src/org/eclipse/core/runtime/SubMonitor.java

```diff
@@ -784,6 +785,38 @@ public SubMonitor newChild(int totalWork) {
 		return newChild(totalWork, SUPPRESS_BEGINTASK);
 	}
 
+	@Override
+	public IProgressMonitor create(int totalWork, ProgressOption... options) {
+		if (options.length == 0) {
+			return newChild(totalWork);
+		}
+		if (options.length == 1 && options[0] == ProgressOption.SUPPRESS_SUBTASK) {
+			return newChild(totalWork, SUPPRESS_BEGINTASK | SUPPRESS_SUBTASK);
+		}
+		// use default implementation for ProgressOption.PREPEND_MAIN_LABEL:
+		return create(this, totalWork, options);
```

Comment:
Can we please just implement `PREPEND_MAIN_LABEL_TO_SUBTASK` in SubMonitor? It should not be to hard to implement and I think such change should prove we do not need to rely on deprecated API and all can be implemented by SubMonitor itself.

Reply:
If anybody wants to add such an optimization that could be done in future commits. I see no value in making the implementation more complicated.